### PR TITLE
feat(footer): connected indicator added under chatbar (AP-356)

### PR DESCRIPTION
**Pull request by jpanay**
**What this PR does** 📖
Shows if peer is connected or disconnected under chatbar
**Which issue(s) this PR fixes** 🔨
AP-356

**Special notes for reviewers** 🗒️

**Additional comments** 🎤

### CODE REVIEW

**Comment by phillsatellite** [comment]
Should indicator appear if user has no friends added?
Screen Shot 2022-02-04 at 2 26 59 PM

**Comment by jpanay** [comment]
No it should not I'll fix that

File: components/views/chat/chatbar/footer/Footer.vue [hunk 1]
```diff
@@ -1,5 +1,24 @@
 <template>
   <div id="bottom-bar-container">
+    <div v-if="!typing" class="is-connected">
```
Comment (WanderingHogan) [inline components/views/chat/chatbar/footer/Footer.vue:3]: Can you make this it's own ui chat component

**Review by WanderingHogan — CHANGES_REQUESTED** [review]
Can you:
1. Split footer into a template, less, and vue file
2. Create a is-connected component
3. Add english to locales (in this repo, not the standalone repo)

File: components/views/chat/chatbar/footer/Footer.vue (hunk 1 above)
Comment (jpanay) [reply components/views/chat/chatbar/footer/Footer.vue]: All requests fixed, bug found by Phill fixed

**Review by molimauro — APPROVED** [review]

**Review by phillsatellite — APPROVED** [review]

**Review by WanderingHogan — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
https://user-images.githubusercontent.com/29093946/153056464-dfa50f04-0f32-4637-b21d-79865944f184.mov

says hogan is not connected on hogan's chat
change chat
still says hogan is not connected until page fully loads

File: components/views/chat/chatbar/footer/is-connected/Connected.html [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    size="1x"
+  />
+  <TypographyText
+    v-if="$Hounddog.getActiveFriend($store.state.friends)"
+    :text="`${$Hounddog.getActiveFriend($store.state.friends).name}`"
+    :size="6"
+  />
+  <TypographyText
+    :text="`is ${
+      $store.state.webrtc.connectedPeer ? 'connected' : 'not connected'
```
Comment (stavares843) [inline components/views/chat/chatbar/footer/is-connected/Connected.html:18]: this should be added to locales the connected/not connected

**Comment by stavares843** [comment]
https://user-images.githubusercontent.com/29093946/153058038-12efeb58-f1e9-48d7-bb63-2897d627b6f2.mov


the connected/not connected updates fine

I closed the tab of user B and updated on user A saying it was no longer connected

but then returned error on user A
Captura de ecrã 2022-02-08, às 19 05 40

and after reloading the page says user B is connected again

Captura de ecrã 2022-02-08, às 19 08 34

but tab of user B was closed and not reopened

**Comment by jpanay** [comment]
> Gravacao.do.ecra.2022-02-08.as.19.03.52.mov 
> the connected/not connected updates fine
> 
> I closed the tab of user B and updated user A saying it was no longer connected
> 
> but then returned error on user A Captura de ecrã 2022-02-08, às 19 05 40
> 
> and after reloading the page says user B is connected again
> 
> Captura de ecrã 2022-02-08, às 19 08 34
> 
> but the tab of user B was closed and not reopened

I'm trying to replicate this but I'm not getting that peer not connected error. I'm opening both tabs, waiting for connection, and closing one, but can't seem to get it to throw that error.

**Comment by jpanay** [comment]
> > Gravacao.do.ecra.2022-02-08.as.19.03.52.mov
> > the connected/not connected updates fine
> > I closed the tab of user B and updated user A saying it was no longer connected
> > but then returned error on user A Captura de ecrã 2022-02-08, às 19 05 40
> > and after reloading the page says user B is connected again
> > Captura de ecrã 2022-02-08, às 19 08 34
> > but the tab of user B was closed and not reopened
> 
> I'm trying to replicate this but I'm not getting that peer not connected error. I'm opening both tabs, waiting for connection, and closing one, but can't seem to get it to throw that error.

Maybe I fixed it inadvertently with a few changes I made, can you test it again

**Comment by phillsatellite** [comment]
I tested again, I got same error as @stavares843 after attempting to message user B after they disconnected, the only thing that was different on mine was when I reloaded page it still said User B was not connected 
Screen Shot 2022-02-10 at 11 47 00 AM

**Comment by WanderingHogan** [comment]
@jpanay The error is here:
image

typingNotifyHandler in the chatbar.vue file.

`const activePeer = this.$WebRTC.getPeer(activeFriend.address)` can come back as undefined

In lieue of digging into getPeer in the webrtc library, for now, please:

In the chatbar.vue file wrap the getPeer call in a try catch, 
Screen Shot 2022-02-10 at 8 01 12 PM

**Comment by WanderingHogan** [comment]
@jpanay I can reproduce the error on my end

**Comment by jpanay** [comment]
> @jpanay I can reproduce the error on my end

Yea I really don't know why I wasn't able to if all three of you can

**Review by WanderingHogan — APPROVED** [review]

**Comment by stavares843** [comment]
Captura de ecrã 2022-02-12, às 02 28 41


i have 2 users, they were both connected, I deactivated internet, and even now both have internet, still says one of them is not connected

File: components/views/chat/chatbar/footer/Footer.less [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#bottom-bar-container {
+  display: flex;
+  justify-content: space-between;
+  // height: 1.5rem;
```
Comment (stavares843) [inline components/views/chat/chatbar/footer/Footer.less:4]: commented-out code 🔨

**Comment by jpanay** [comment]
> Captura de ecrã 2022-02-12, às 02 28 41
> 
> i have 2 users, they were both connected, I deactivated internet, and even now both have internet, still says one of them is not connected
> 
> full flow was
> 
> * user A added user B
> * user B sent message to user A
> * user A sent message to user B
> * go to console network settings on user A and select offline
> 
> Captura de ecrã 2022-02-12, às 02 30 12
> 
> * user B sent message to user A - it sends
> * user A sent message to user B - doesn't send, there's no internet on that user
> * remove offline setting
> * user A sent message to user B - sends successfully
> * still shows user B not connected on user A side

Could you make a separate ticket for this, it's a webrtc issue beyond the scope of this PR

**Comment by stavares843** [comment]
the ticket which this tickets is expected to implement says the following:

''Show an indication under the chat input if the peer is connected/disconnected, might require checking/watching for events that don't exist?''

so, one of those events, would be a disconnected event as well, such as without internet

is not outside the scope, because the PR should include that doesn't matter if is or not a webrtc issue

but let me know cc @WanderingHogan

**Comment by WanderingHogan** [comment]
I think that this is a problem with devtools in chrome - i dont think all connection types are getting closed when you disable the connection.

This acts differently when you are on two different devices. Eventually the webrtc connection times out and shows both users are offline. I think that there are problems managing the webrtc connection that should be included in future tickets but we need to debug some things first. 

If this ticket works otherwise and doesn't cause any crashes i think it can go in.

**Review by stavares843 — APPROVED** [review]

**Review by stavares843 — APPROVED** [review]

**Pull request merged**